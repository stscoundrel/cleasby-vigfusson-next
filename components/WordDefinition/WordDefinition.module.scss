.section {
  padding: 20px 0;
}

.definitionList {
  margin-bottom: 20px;
}

.itemDescription {
  padding-left: 20px;
}

@media only screen and (min-width: 1200px) {
  .section {
    max-width: 740px;
  }
}