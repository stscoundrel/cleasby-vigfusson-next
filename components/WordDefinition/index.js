export default function WordDefinition({ data }) {
  const { word, definitions } = data

  return (
    <article>
      <header>
        <h1 lang="non">{word}</h1>
        <p><dfn>{word}</dfn> Old Norse word can mean:</p>
      </header>
      {definitions.map((definition, index) => (
        <dl key={`definition-${index}`}>
          <dt><strong>{word}</strong></dt>
          <dd dangerouslySetInnerHTML={ { __html: definition } }></dd>
        </dl>
      ))}
    </article>
  )
}
