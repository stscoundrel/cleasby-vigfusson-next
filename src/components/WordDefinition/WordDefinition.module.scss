.section {
  padding: 20px 0;
}

.subHeading {
  display: block;
  margin-bottom: 30px;
}

.rune {
  display: block;
  margin-top: 10px;
  font-size: 20px;
}

.definitionList {
  margin-bottom: 20px;
}

.itemDescription {
  padding-left: 20px;
}

@media only screen and (min-width: 768px) {
  .rune {
    display: inline-block;
    font-size: 16px;
    margin-top: 0;
    margin-left: 5px;
  }
}

@media only screen and (min-width: 1200px) {
  .section {
    max-width: 740px;
  }
}