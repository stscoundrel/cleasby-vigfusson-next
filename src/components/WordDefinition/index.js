import { capitalize } from 'lib/utils/strings'
import { lettersToRunes } from 'younger-futhark'
import styles from './WordDefinition.module.scss'

export default function WordDefinition({ data }) {
  const { word, definitions } = data

  return (
    <article className={styles.section}>
      <header>
        <h1 className="capitalize" lang="non">{capitalize(word)}</h1>
        <p>Possible runic inscription in <em>Younger Futhark:</em> { lettersToRunes(word) }</p>
        <p>Meaning of Old Norse word <em>&quot;{word}&quot;</em></p>
      </header>

      {definitions.length > 1 && <p><dfn className="capitalize">{word}</dfn> Old Norse word can mean:</p>}
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong></dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={ { __html: definition } }
          ></dd>
        </dl>
      ))}
    </article>
  )
}
