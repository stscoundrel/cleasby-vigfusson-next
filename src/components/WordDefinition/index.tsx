import { youngerFuthark } from 'riimut'
import type { Crosslink } from 'scandinavian-dictionary-crosslinker'
import { capitalize, getOlderSpelling } from 'lib/utils/strings'
import { addAbbreviationsToContent, type CombinedAbbreviations } from 'lib/services/abbreviations'
import Abbreviations from 'components/Abbreviations'
import type { DictionaryEntry } from 'lib/services/dictionary'
import Crosslinks from 'components/Crosslinks'
import SimilarEntries from 'components/SimilarEntries'
import styles from './WordDefinition.module.scss'

interface WordDefinitionProps{
  entry: DictionaryEntry,
  similarEntries: DictionaryEntry[],
  abbreviations: CombinedAbbreviations,
  crosslinks: Crosslink[],
}

export default function WordDefinition({
  entry, similarEntries, abbreviations, crosslinks,
}: WordDefinitionProps) {
  const { word, definitions } = entry
  const olderForm = getOlderSpelling(word)
  const hasOlderForm = word !== olderForm

  return (
    <article className={styles.section}>
      <header>
        <h1 lang="non">{capitalize(word)}</h1>

        <small className={styles.subHeading}>
          Old Norse Dictionary - {word.toLowerCase()}
        </small>
        <p>Meaning of Old Norse word <em>&quot;{word}&quot;</em>
        {hasOlderForm && <> (or <em>{olderForm}</em>)</>} in English.</p>
      </header>

      <p>As defined by the Cleasby & Vigfusson Old Norse to English dictionary:</p>

      {definitions.length > 1 && <p><dfn className="capitalize">{word}</dfn> Old Norse word can mean:</p>}
      {definitions.map((definition, index) => (
        <dl className={styles.definitionList} key={`definition-${index}`}>
          <dt><strong>{word}</strong> {hasOlderForm && <>({olderForm})</>}</dt>
          <dd
            className={styles.itemDescription}
            dangerouslySetInnerHTML={{
              __html: addAbbreviationsToContent(definition, abbreviations),
            } }
          ></dd>
        </dl>
      ))}

      {hasOlderForm
        && <p>
          <strong>Orthography: </strong>The Cleasby & Vigfusson book used letter <em>ö </em>
          to represent the original Old Norse vowel <em>ǫ</em>. Therefore, <em>{word}</em> may be
          more accurately written as <em>{olderForm}</em>.
        </p>}

      <p>Possible runic inscription in <em>Younger Futhark</em>:
        <span className={styles.rune}>{ youngerFuthark.lettersToRunes(word) }</span><br />
      <small>Younger Futhark runes were used from 8th to 12th centuries
        in Scandinavia and their overseas settlements</small>
      </p>

      <SimilarEntries entries={similarEntries} />
      <Abbreviations abbreviations={abbreviations} />
      <br />
      <Crosslinks crosslinks={crosslinks} />
    </article>
  )
}
