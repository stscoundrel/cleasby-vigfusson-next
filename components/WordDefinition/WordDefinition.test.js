import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import WordDefinition from './index'

const word = {
  word: 'af-burðr',
  definitions: [
    'm. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Orkn. 464.',
    '<strong>II.</strong> gen. sing. and pl. <strong>afburðar-, a-,</strong> freq. used as a prefix in some COMPDS with the notion of <i>gloriously, with distinction.</i> <strong>afburðar-digr,</strong> adj. <i>very thick</i>, biðr. 24. <strong>afburða-fræknligr,</strong> adj. <i>very gallant,</i> Ísl. ii. 369. <strong>afburðar-járn,</strong> n. <i>excellent iron,</i> Fms. x. 173. <strong>afburðar-maðr,</strong> m. <i>a man of mark,</i> Rb. 316, Orkn. 474, Grett. 133, Finnb. 318. <strong>afburðar-mikill,</strong> adj. <i>conspicuous,</i> Fms. v. 181. <strong>afburðar-skip,</strong> n. <i>a fine ship,</i> Fas. iii. 106. <strong>afburðar-vel,</strong> adv. <i>very well</i>, Hkr. ii. 265, Fms. ix. 515. <strong>afburðar-vænn,</strong> adj. <i>very fine,</i> Fas. i. 182.',
  ],
  slug: 'af-burdr',
}

describe('WordDefinition component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<WordDefinition data={word} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordDefinition data={word} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(<WordDefinition data={word} />)
    const { root } = tree

    expect(root.findByType('h1').children).toEqual(['Af-burðr'])
  })

  test('Has correct amount of definitions', () => {
    const tree = renderer.create(<WordDefinition data={word} />)
    const { root } = tree

    expect(root.findAllByType('dt').length).toEqual(2)
  })
})
