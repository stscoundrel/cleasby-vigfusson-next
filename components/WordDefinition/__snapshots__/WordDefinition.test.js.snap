// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordDefinition component Matches snapshot 1`] = `
<article>
  <header>
    <h1
      lang="non"
    >
      af-burðr
    </h1>
    <p>
      <dfn>
        af-burðr
      </dfn>
       Old Norse word can mean:
    </p>
  </header>
  <dl>
    <dt>
      <strong>
        af-burðr
      </strong>
    </dt>
    <dd
      dangerouslySetInnerHTML={
        Object {
          "__html": "m. (also spelt abb-), <i>odds, balance, bias, success</i> (cp. bera af, <i>to prevail</i>); kvað honum eigi annat vænna til afburðar, <i>in order to get the better of it,</i> Sd. 166; sá hann at engi varð afburðrinn, <i>they fought ‘aequo Marte,’</i> Sturl. ii. 74; hann ætlaði sér afburð, <i>he meant to keep the odds in his own hand,</i> Ísl. ii. 450; skal nú faraí haustvíking, ok vilda ek, at hon yrði eigi með minnum afburðum, <i>less glorious,</i> Orkn. 464.",
        }
      }
    />
  </dl>
  <dl>
    <dt>
      <strong>
        af-burðr
      </strong>
    </dt>
    <dd
      dangerouslySetInnerHTML={
        Object {
          "__html": "<strong>II.</strong> gen. sing. and pl. <strong>afburðar-, a-,</strong> freq. used as a prefix in some COMPDS with the notion of <i>gloriously, with distinction.</i> <strong>afburðar-digr,</strong> adj. <i>very thick</i>, biðr. 24. <strong>afburða-fræknligr,</strong> adj. <i>very gallant,</i> Ísl. ii. 369. <strong>afburðar-járn,</strong> n. <i>excellent iron,</i> Fms. x. 173. <strong>afburðar-maðr,</strong> m. <i>a man of mark,</i> Rb. 316, Orkn. 474, Grett. 133, Finnb. 318. <strong>afburðar-mikill,</strong> adj. <i>conspicuous,</i> Fms. v. 181. <strong>afburðar-skip,</strong> n. <i>a fine ship,</i> Fas. iii. 106. <strong>afburðar-vel,</strong> adv. <i>very well</i>, Hkr. ii. 265, Fms. ix. 515. <strong>afburðar-vænn,</strong> adj. <i>very fine,</i> Fas. i. 182.",
        }
      }
    />
  </dl>
</article>
`;
